contracts/HtsSystemContract.sol

```diff
@@ -395,6 +401,11 @@ contract HtsSystemContract is IHederaTokenService, IERC20Events, IERC721Events {
         assembly { amount := sload(slot) }
     }
 
+    function __tokenURI(uint256 serialId) private returns (string memory uri) {
+        bytes32 slot = _tokenUriSlot(uint32(serialId));
+        uri = readString(slot);
```

Comment:
Let's see if this works, it points `_uri` to the dynamic slot so we can read using usual Solidity.

```suggestion
        string storage _uri;
        assembly { _uri.slot := slot }
        uri = _uri;
```

Reply:
Great suggestion! Yes, it works!